endurance-page-cache.php

```diff
@@ -585,19 +595,50 @@ public function purge_cdn_single( $pattern = '' ) {
 		/**
 		 * Ensure that a URI isn't purged more than once per minute.
 		 *
-		 * @param string $value URI being purged
+		 * @param string $uri URI being purged
 		 *
-		 * @return bool
+		 * @return bool True if additional purges should be avoided, false otherwise.
 		 */
-		public function purge_throttle( $value ) {
-			$purged = get_transient( 'epc_purged_' . md5( $value ) );
-			if ( ( true === $purged || in_array( md5( $value ), $this->purged, true ) ) && false === $this->force_purge ) {
-				return true;
+		public function should_throttle( $uri ) {
+
+			$should_throttle = false;
+
+			$this->should_update_purges = true;
+
+			$hash = md5( $uri );
+
+			if ( array_key_exists( $hash, $this->purged ) ) {
```

Comment:
It's kinda a micro-optimization, but `isset()` is [typically faster](https://gist.github.com/alcaeus/536156663fac96744eba77b3e133e50a) than `array_key_exists()` and spreading that across the farm and how frequently this function is called will add up.